packages/core/src/node/serve.ts

```diff
@@ -22,35 +14,31 @@ export async function serve(options: ServeOptions) {
   const { config, port: userPort, host: userHost } = options;
   const envPort = process.env.PORT;
   const envHost = process.env.HOST;
-  const port = envPort || userPort || 4173;
-  const host = envHost || userHost || 'localhost';
-  const base = config?.base?.replace(/^\//, '').replace(/\/$/, '') || '';
-  const compress = compression();
-  const serve = sirv(config?.outDir || OUTPUT_DIR, {
-    etag: true,
-    maxAge: 31536000,
-    immutable: true,
+  const { builderConfig } = config;
+  const port = Number(
+    envPort || userPort || builderConfig?.server?.port || 4173,
+  );
+  const host =
+    envHost || userHost || builderConfig?.server?.host || 'localhost';
+
+  config.builderConfig = mergeRsbuildConfig(builderConfig, {
+    server: {
+      port,
+      host,
+    },
   });
 
-  if (base) {
-    polka()
-      .use(base, compress, serve)
-      .listen(port, host, (err: Error) => {
-        if (err) {
-          throw err;
-        }
-        console.log(
-          `Preview server running at http://${host}:${port}/${base}/\n`,
-        );
-      });
-  } else {
-    polka()
-      .use(compress, serve)
-      .listen(port, host, (err: Error) => {
-        if (err) {
-          throw err;
-        }
-        console.log(`Preview server running at http://${host}:${port}/\n`);
-      });
-  }
+  const pluginDriver = new PluginDriver(config, true);
+  await pluginDriver.init();
+
+  const modifiedConfig = await pluginDriver.modifyConfig();
+
+  const builder = await initRsbuild(
+    config.root,
+    modifiedConfig,
+    pluginDriver,
+    false,
+  );
+
+  await builder.preview();
```

Comment:
Could the server take the `base` path in preview log?

Reply:
Yes, `base` is supported in `createInternalBuildConfig`, we do not need to add extra code here 😄

![image](https://github.com/web-infra-dev/rspress/assets/7237365/dee2169b-84eb-41b1-9f99-3461b179ef8c)

Reply:
Awesome